orttraining/orttraining/training_ops/cpu/nn/broadcast_grad_args.cc

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "orttraining/training_ops/cpu/nn/broadcast_grad_args.h"
+
+namespace onnxruntime {
+namespace contrib {
+#define REGISTER_KERNEL_TYPED(T)                                  \
+  ONNX_OPERATOR_TYPED_KERNEL_EX(                                  \
+      BroadcastGradientArgs,                                      \
+      kMSDomain,                                                  \
+      1,                                                          \
+      T,                                                          \
+      kCpuExecutionProvider,                                      \
+      KernelDefBuilder()                                          \
+          .TypeConstraint("T", DataTypeImpl::GetTensorType<T>()), \
+      BroadcastGradientArgs<T>);
+
+REGISTER_KERNEL_TYPED(int64_t)
+
+template <typename T>
+Status BroadcastGradientArgs<T>::Compute(OpKernelContext* context) const {
+  const Tensor* a_shape = context->Input<Tensor>(0);
+  const Tensor* b_shape = context->Input<Tensor>(1);
+  const T* A_dims = a_shape->template Data<T>();
+  const T* B_dims = b_shape->template Data<T>();
+
+  const int a_size = a_shape->Shape().Size();
```

Comment:
Windows compile error: warning C4244: 'initializing': conver
sion from 'int64_t' to 'int', possible loss of data